AsyncAwait.Tasks/AsyncAwait.Task2.CodeReviewChallenge/Middleware/StatisticMiddleware.cs

```diff
@@ -24,16 +24,13 @@ public async Task InvokeAsync(HttpContext context)
         string path = context.Request.Path;
 
         var staticRegTask = Task.Run(
-            () => _statisticService.RegisterVisitAsync(path)
-                .ConfigureAwait(false)
-                .GetAwaiter().OnCompleted(UpdateHeaders));
-        Console.WriteLine(staticRegTask.Status); // just for debugging purposes
+           async () => _statisticService.RegisterVisitAsync(path).GetAwaiter().OnCompleted(UpdateHeaders));

```

Comment:
Why OnCompleted is needed here? I suggest just using await for all async methods, something like this
`// Register visit asynchronously and await for completion
        await _statisticService.RegisterVisitAsync(path).ConfigureAwait(false);
        // Update headers asynchronously and await for completion
        long visitsCount = await _statisticService.GetVisitsCountAsync(path).ConfigureAwait(false);
        context.Response.Headers.Add(CustomHttpHeaders.TotalPageVisits, visitsCount.ToString());
`